search-algos/linear_search/l_search.c

```diff
@@ -2,29 +2,31 @@
 #include<stdio.h>
 void main()
 {
-   int i,n,x,flag=0;
-   printf("enter how many value in array\n");  \\Taking no of elements in an array from user
-   scanf("%d",&n);
-   int a[n];
+   int flag,n,x;
```

Comment:
Initialize the value of flag with 0, as you are using zero as the case when the element do not exit.